### PR TITLE
#2084 delete unit column

### LINKED ISSUE #2084: Delete Unit Frontend
### Description

We want to add a tool to visualize and allow deleting of units to the admin tools page. You should create a table of units (the same as the rest of the tables we have on that page for manufacturers/ material types). You should also add a column to this table with trash can icons that allow you to delete a unit

### Acceptance Criteria

- [ ] There is a delete unit tool on the admin tools page in the same style as the manufacturer/ material type tables
- [ ] This tool should go under the BOM config page block

### Proposed Solution

Take a look a code that's doing similar things. You'll need to add an API call/ hook to use the delete unit endpoint that's been added

### PR DESCRIPTION
**Pull request by anyadas11**
## Changes

Added a column of trash can icons in the unit table in admin tools that allow for deleting a unit. 

## Notes

- there is an error occurring and I'm not sure where it's coming from

## Screenshots
Screenshot 2024-03-29 at 4 21 08 PM
Screenshot 2024-03-29 at 4 21 26 PM

## To Do

- [ ] fix delete functionality  - figure out the error with handleDeleteUnit

## Checklist

- [X] All commits are tagged with the ticket number
- [X] All code follows repository-configured prettier formatting
- [X] No merge conflicts
- [X] Screenshots of UI changes (see Screenshots section)
- [X] Remove any non-applicable sections of this template
- [X] Assign the PR to yourself
- [X] No `yarn.lock` changes (unless dependencies have changed)
- [X] Request reviewers & ping on Slack
- [X] PR is linked to the ticket (fill in the closes line below)

Closes #2084

### CODE REVIEW

File: src/frontend/src/pages/AdminToolsPage/AdminToolsBOMConfig.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { Box } from '@mui/system';
 
 const AdminToolsBOMConfig: React.FC = () => {
   return (
-    <Box>
+    <Box paddingBottom="4">
```
Comment (bderbs30) [inline src/frontend/src/pages/AdminToolsPage/AdminToolsBOMConfig.tsx:9]: The padding at the bottom isn't working 
image
Look at the syntax it should be 

paddingBottom={4}

**Comment by bderbs30** [comment]
image
In the top right of the table the edges aren't bordered by a black line like the others

**Comment by bderbs30** [comment]
> image In the top right of the table the edges aren't bordered by a black line like the others


You'll want to look at this code specifically in the columns prop of the admin tool table
image

The change you'll need to make is present in this code example

image


if you have any questions let me know

File: src/backend/src/controllers/projects.controllers.ts [hunk 2]
```diff
@@ -246,8 +246,8 @@ export default class ProjectsController {
   static async deleteUnit(req: Request, res: Response, next: NextFunction) {
     try {
       const user: User = await getCurrentUser(res);
-      const { unitName } = req.params;
-      const deletedUnit = await ProjectsService.deleteUnit(user, unitName);
+      const { unitId } = req.params;
```
Comment (RChandler234) [inline src/backend/src/controllers/projects.controllers.ts:249]: this should still be unitName, that's what the field is called in the schema

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by bderbs30 — APPROVED** [review]
LGTM 💸

**Pull request merged**